### DISCUSSION TITLE
Query for emit function of pymeasure

### DISCUSSION THREAD

**Discussion by AdityaOP** [pymeasure/pymeasure#415] [Q&A]
Hello Everyone,
From my experiment I am getting 2 arrays. Now I want to emit this array to csv file through self.emit function. I try to emit this array with the help of for loop by sending 1 value at a time. By this method the data is emitting correctly but taking too much time as I have array of 10,000 values. If i directly emit the arrays than there is only one row in csv file containing both arrays which is not correct. Can I know is there any way I can send data correctly and faster? 
Thank You,
Aditya

**Comment by msmttchr** [comment]
I am not sure whether you're using a `ManagedWindow`, if this is the case, it is possible you are slow down by graph output.
Have you tried to run your experiment with a batch script like `script.py` in `examples` directory?

**Comment by moritzj29** [comment]
we discussed together in this PR draft, so I thought I reference it for future readers: https://github.com/pymeasure/pymeasure/pull/277

This is a current limitation of pymeasure requiring to write only one value per emit call. This is very limiting for spectral data or scope traces (many datapoints returned from one instrument read).

please keep in mind that the PR is still a draft and not available in the release (you can pull it manually though)
